receiver/prometheusreceiver/internal/metricsbuilder.go

```diff
@@ -140,7 +138,7 @@ func (b *metricBuilder) AddDataPoint(ls labels.Labels, t int64, v float64) error
 func (b *metricBuilder) Build() ([]*metricspb.Metric, int, int, error) {
 	if !b.hasData {
 		if b.hasInternalMetric {
-			return dummyMetrics, 0, 0, nil
+			return make([]*metricspb.Metric, 0), 0, 0, nil
```

Comment:
Can this return nil instead of a slice with len 0?

Reply:
I'm not familiar enough with the semantics enough here to be confident returning nil and/or an error will be correct behavior. Leaving a TODO.

Reply:
I don't see the todo, did you forget to push the commit?

Reply:
I will do it.